morphocut/pipeline/annotation.py

```diff
@@ -34,43 +38,51 @@ def __call__(self, input=None):
                 disk(radius))
 
             # Draw contours
-            _, contours, _ = cv.findContours(img_as_ubyte(dilated_mask), 1, 2)
+            #_, contours, _ = cv.findContours(img_as_ubyte(dilated_mask), 1, 2)
+            boundary_mask = skimage.segmentation.find_boundaries(dilated_mask, mode="outer")
 
-            contour_image = img_as_ubyte(img, True)
+            contour_image = img_as_ubyte(image, True)
             color = (0, 255, 0)
 
-            cv.drawContours(contour_image, contours, -1, color, 1)
+            #cv.drawContours(contour_image, contours, -1, color, 1)
+            contour_image[boundary_mask] = color
 
-            obj["facets"][self.output_facet] = {
-                "image": contour_image
-            }
+            #obj["facets"][self.output_facet] = {
+            #    "image": contour_image
+            #}
 
-            yield obj
+            yield self.prepare_output(obj, contour_image)
 
 
-class DrawContoursOnParent(NodeBase):
+class DrawContoursOnParent(Node):
 
-    def __init__(self, image_facet, mask_facet, output_facet, dilate_rel=0, dilate_abs=0):
-        self.image_facet = image_facet
-        self.mask_facet = mask_facet
-        self.output_facet = output_facet
+    def __init__(self, image, mask, output, dilate_rel=0, dilate_abs=0):
+        self.image = image
+        self.mask = mask
+        self.output = output
         self.dilate_rel = dilate_rel
         self.dilate_abs = dilate_abs
 
-    def __call__(self, input=None):
-        for obj in input:
-            parent, parent_slice = obj["parent"], obj["parent_slice"]
+    def transform_stream(self, stream):
+        for obj in stream:
+            parent, parent_slice = self.prepare_input(obj, ("parent", "parent_slice"))
```

Comment:
Where do you get the `parent_slice` from, currently? This needs to be supplied as a parameter to `__init__`.